File: src/API/Modules/Shows/Cinemanjaro.Shows.Domain/Aggregates/Show.cs
using Cinemanjaro.Shows.Domain.Entities;
using Cinemanjaro.Shows.Domain.Exceptions;
using Cinemanjaro.Shows.Domain.ValueObjects;
using MongoDB.Bson;

namespace Cinemanjaro.Shows.Domain.Aggregates
{
    public class Show
    {
        public ObjectId Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public IEnumerable<Seat> Seats { get; set; }

        public void BookSeats(IEnumerable<SeatPosition> seatPositions)
        {
            foreach (SeatPosition position in seatPositions)
            {
                var seat = Seats.FirstOrDefault(x => x.Position == position);
                if (seat == null)
                    throw new SeatDoesNotExistException(position);

                seat.BookSeat();
            }
        }
    }
}

File: src/APP/cinemanjaro-app/src/app/models/Show.ts
import { Seat, SeatPosition } from "./Seat";

export interface Show{
    id : string,
    date : Date,
    title : string,
    iconURL : string,
    movieId : string,
    lengthMins : number,
    genres : string[],
}

export interface ShowsResponse{
    shows:Show[]
}

export interface ShowDetails{
    id : string,
    date : Date,
    title : string,
    iconURL : string,
    movieId : string,
    lengthMins : number,
    genres : string[],
    seats : Seat[],
}

export interface SeatsToBook{
    email : string,
    seatPositions: SeatPosition[],
}